### ISSUE TITLE
Insert ALB between client and server

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#553]
Create an Application Load Balancer and insert it between the client and the server

This is a step toward zero downtime deploys (#550). I chose to make it a separate issue rather than a check-box, because I'm experimenting with participation in the Albuquerque Game Developers Group Accountability Team.

**Comment by ctm** [comment]
I've used ACM to get certs for `devctm.com`, `assets.devctm.com` (which I don't need) and `ws.devctm.com`. However, there are a few configuration settings that I'm not sure what values to use. I'll have to start over and make a list. Grr...

**Comment by ctm** [comment]
I got the load balancer created and managed to get it configured to automatically redirect http on port 80 to https on 443, but  when I tried to insert the ALB in the `ws.devctm.com` path, mb2 stopped working for me.

FTR, I inserted it as `ws.devctm.com`, because that's supposed to route directly to my instance, rather than through Cloudflare. As such, I considered it easier. However, I don't really fully appreciate how an `https` request gets upgraded to `wss`, so that might have been a problem.

Sometime "soon" I can try simply doing `devctm.com`, which I think I'll have to turn into a `CNAME` to insert the ALB. If that works, great. If it doesn't, I can take Cloudflare out of the loop, test that and then try inserting the ALB.

**Comment by ctm** [comment]
Turns out I had never supplied a target to the `devctm` target group. I'm a little surprised that nothing in the AWS console pointed that out to me, but once I started looking at all the configuration settings, that became apparent.

I now have `ws.devctm.com` running through the load balancer, which is enough to close this issue. Whee!

**Comment by ctm** [comment]
I'm leaving this closed, but I did take the ALB out from in front of `ws.devctm.com` after people (including myself) having tables closing spontaneously. I'll do some research, but considering the trouble I had routing the WebSockets through CloudFlare, my guess is the ALB is causing trouble. The good news is I really only need the ALB for `devctm.com` and I only chose to insert it before `ws.devctm.com` due to wanting to avoid CloudFlare when I was experimenting. OTOH, I'll still have to experiment with putting it before Cloudflare, but that can be done as a separate (or part of another) issue.
